pkg/cloudinit/layout_plugin.go

```diff
@@ -111,7 +102,16 @@ func layoutPlugin(l logger.Interface, s schema.Stage, fs vfs.FS, console plugins
 				return fmt.Errorf("Formatting partition failed: %s\nError: %w", out, err)
 			}
 		}
-		return nil
 	}
+
+	if s.Layout.Expand != nil {
+		l.Infof("Extending last partition up to %d MiB", s.Layout.Expand.Size)
+		out, err := dev.ExpandLastPartition(s.Layout.Expand.Size)
+		if err != nil {
+			l.Error(out)
+			return err
+		}
+	}
```

Comment:
NOTE: Let's try an installation without custom partitions with this fix to see if this breaks something. @Itxaka has a hunch...